torch_xla/core/dynamo_bridge.py

```diff
@@ -307,8 +309,32 @@ def extract_internal(xla_model: torch.fx.GraphModule):
   # should be removed to avoid extra computation executed and in place updates op
   # mistakenlly update the input tensors.
   torch_xla._XLAC._clear_pending_irs(str(xm.xla_device()))
+  return (xla_args_sharding_spec, args_and_out, graph_hash,
+          arg_index_to_need_update_index, none_remover, graph_input_matcher,
+          dumb_return_handler, xla_args_need_update)
+
+
+def extract_internal(xla_model: torch.fx.GraphModule):
+  (xla_args_sharding_spec, args_and_out, graph_hash,
+   arg_index_to_need_update_index, none_remover, graph_input_matcher,
+   dumb_return_handler, xla_args_need_update) = extract_graph_helper(xla_model)
 
   def optimized_mod(*args):
+    nonlocal xla_args_sharding_spec
+    nonlocal args_and_out
+    nonlocal graph_hash
+    nonlocal arg_index_to_need_update_index
+    nonlocal none_remover
+    nonlocal graph_input_matcher
+    nonlocal dumb_return_handler
+    nonlocal xla_args_need_update
+    print(xla_args_sharding_spec)
+    current_arg_sharding_spec = torch_xla._XLAC._get_xla_sharding_specs(args)
+    if current_arg_sharding_spec != xla_args_sharding_spec:
+      (xla_args_sharding_spec, args_and_out, graph_hash,
+      arg_index_to_need_update_index, none_remover, graph_input_matcher,
+      dumb_return_handler, xla_args_need_update) = extract_graph_helper(xla_model)
```

Comment:
now I think about it, I am not sure if `args` that has the new sharding info will be updated in the `xla_model.xla_args`, need to verify that tmr.

Reply:
this is fixed, we need to update `xla_model.xla_arg`